p2p/src/transport.rs

```diff
@@ -0,0 +1,57 @@
+use std::fmt;
+use std::io::{Read, Write};
+use std::net::SocketAddr;
+
+use tendermint::public_key::PublicKey;
+
+use eyre::Result;
+
+pub struct BindInfo {
+    pub addr: SocketAddr,
+    pub advertise_addrs: Vec<SocketAddr>,
+    pub public_key: PublicKey,
+}
+
+pub struct ConnectInfo {}
+
+#[derive(Clone, Copy, Hash, Eq, PartialEq)]
+pub enum StreamId {
+    Pex,
+}
+
+pub enum Direction<Conn> {
+    Incoming(Conn),
+    Outgoing(Conn),
+}
+
+pub trait Connection: Send {
+    type Error: 'static + fmt::Display + std::error::Error + Send + Sync;
```

Comment:
The `fmt::Display` bound is redundant here as it's [already required](https://doc.rust-lang.org/stable/std/error/trait.Error.html) by the `std::error::Error` trait.